WordPress/src/main/java/org/wordpress/android/ui/reader/views/ReaderWebView.java

```diff
@@ -188,6 +207,34 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
                 return false;
             }
         }
+
+        @Override
+        public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
+            // Intercept requests and add the WP.com authorization header
+            if (!TextUtils.isEmpty(mToken) && isImageUrl(url)) {
```

Comment:
Sounds good, I didn't know we had the ability to tell if the post is private :+1: